pkg/replication/replication.go

```diff
@@ -305,6 +315,7 @@ func (r *replication) handleRenewalErrors(session kp.Session, renewalErrCh chan
 	defer func() {
 		close(r.quitCh)
 		close(r.errCh)
+		r.rateLimiter.Stop()
```

Comment:
Can you guarantee that all workers (those created in [L219](https://github.com/square/p2/pull/630/files#diff-7ead7087026253e988173f0362b9bd81L219)) have started by the time you stop the timer? If not, they never will.